explorer/interpreter/action_stack.cpp

```diff
@@ -192,8 +194,6 @@ auto ActionStack::Spawn(std::unique_ptr<Action> child, RuntimeScope scope)
 auto ActionStack::ReplaceWith(std::unique_ptr<Action> replacement)
     -> ErrorOr<Success> {
   std::unique_ptr<Action> old = todo_.Pop();
-  CARBON_CHECK(replacement->kind() == old->kind())
```

Comment:
I'm kind of wary of removing this entirely; without looking too far, is this something that we could check that old->kind is a witness if it's changing? (is that going to be restricted in that way?)

Reply:
Done.